go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/abcxyz/abc
 go 1.21
 
 require (
+	github.com/Masterminds/semver/v3 v3.2.1
```

Comment:
Can you share more about the decision to use this library over https://pkg.go.dev/golang.org/x/mod/semver?

Reply:
Good question. I could go either way on this. Here's my reasoning:

 - The Masterminds one is more popular (1200ish imports vs 700ish)
 - The x/mod one has a 0.x version number, which indicates to me that it might not be a stable foundation to build on. The Masterminds one has a 3.x version number.
 - The Masterminds one has an API that seems more sensible to me. It exposes the parsed form of a semver as a struct. The x/mod library which deals with strings and reparses them into a unexported struct every time they're seen (so a sort comparator has to re-parse the semver for every comparison). IMO there's a nice improvement to readability and maintainability to have a SemVer struct that uses the Go type system to guarantee that this semver string is known to be valid. In contrast, the x/mod one just returns empty string when parsing fails (which is the "in-band error" antipattern)

Reply:
Those all sound like really compelling reasons! I'm struggling to think of a good place to capture this decision somewhere in the code.

Reply:
Yeah... I couldn't think of a good way to document it that would actually be helpful to anyone.